src/app/clusters/scenes-server/scenes-server.cpp

```diff
@@ -0,0 +1,993 @@
+
+    // Response data
+    response.groupID = req.groupID;
+    response.sceneID = req.sceneID;
+
+    // Verify Endpoint in group
+    VerifyOrReturn(nullptr != groupProvider);
+    if (0 != req.groupID &&
+        !groupProvider->HasEndpoint(ctx.mCommandHandler.GetAccessingFabricIndex(), req.groupID, ctx.mRequestPath.mEndpointId))
+    {
+        response.status = to_underlying(Protocols::InteractionModel::Status::InvalidCommand);
+        ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+        return;
+    }
+
+    SceneTableEntry scene;
+
+    //  Gets the scene form the table
+    ReturnOnFailure(
+        AddResponseOnError<ResponseType>(ctx, response,
+                                         sceneTable->GetSceneTableEntry(ctx.mCommandHandler.GetAccessingFabricIndex(),
+                                                                        SceneStorageId(req.sceneID, req.groupID), scene)));
+
+    // Verifies how to convert transition time
+    if (scene.mStorageData.mSceneTransitionTimeMs)
+    {
+        if (Commands::ViewScene::Id == ctx.mRequestPath.mCommandId)
+        {
+            response.transitionTime.SetValue(static_cast<uint16_t>(scene.mStorageData.mSceneTransitionTimeMs / 1000));
+        }
+        else if (Commands::EnhancedViewScene::Id == ctx.mRequestPath.mCommandId)
+        {
+            response.transitionTime.SetValue(static_cast<uint16_t>(scene.mStorageData.mSceneTransitionTimeMs / 100));
+        }
+    }
+
+    // Verify if name was saved in the scene
+    if (scene.mStorageData.mNameLength)
+    {
+        response.sceneName.SetValue(CharSpan(scene.mStorageData.mName, scenes::kSceneNameMaxLength));
+    }
+
+    // Response Extension Field Sets buffer
+    Structs::ExtensionFieldSet::Type responseEFSBuffer[scenes::kMaxClustersPerScene];
+    uint8_t deserializedEFSCount = 0;
+
+    // Adds extension field sets to the scene
+    for (uint8_t i = 0; i < scene.mStorageData.mExtensionFieldSets.GetFieldSetCount(); i++)
+    {
+        // gets data from the field in the scene
+        ExtensionFieldSet tempField;
+        scene.mStorageData.mExtensionFieldSets.GetFieldSetAtPosition(tempField, i);
+        ByteSpan efsSpan(tempField.mBytesBuffer, tempField.mUsedBytes);
+
+        // This should only find one handle per cluster
+        for (auto & handler : sceneTable->mHandlerList)
+        {
+            if (handler.SupportsCluster(ctx.mRequestPath.mEndpointId, tempField.mID))
+            {
+                ReturnOnFailure(AddResponseOnError<ResponseType>(
+                    ctx, response,
+                    handler.Deserialize(ctx.mRequestPath.mEndpointId, tempField.mID, efsSpan, responseEFSBuffer[i])));
+                deserializedEFSCount++;
+                break;
+            }
+        }
+    }
+
+    Span<Structs::ExtensionFieldSet::Type> responseEFSSpan(responseEFSBuffer, deserializedEFSCount);
+    response.extensionFieldSets.SetValue(responseEFSSpan);
```

Comment:
This is going to (incorrectly, I believe) send an extensionFieldSets together with an error status if we take our `ReturnOnFailure` early returns.

Put another way, this line should be right before/after we set response.status to success.

And I would think we should set sceneName down there too?